File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/DeviceMetric.ts
import { CodeableConcept } from "./CodeableConcept";
import { Timing } from "./Timing";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type DeviceMetricCalibration = BackboneElement & {
    time?: string;
    type?: string;
    state?: string;
};

export type DeviceMetric = DomainResource & {
    category: string;
    measurementPeriod?: Timing;
    color?: string;
    parent?: Reference;
    unit?: CodeableConcept;
    type: CodeableConcept;
    source?: Reference;
    identifier?: Identifier[];
    calibration?: DeviceMetricCalibration[];
    operationalStatus?: string;
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/DeviceMetric.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Timing
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class DeviceMetricCalibration(BackboneElement):
    time: Optional[str] = None
    type: Optional[str] = None
    state: Optional[str] = None

@dataclass(kw_only=True)
class DeviceMetric(DomainResource):
    category: str
    measurement_period: Optional[Timing] = None
    color: Optional[str] = None
    parent: Optional[Reference] = None
    unit: Optional[CodeableConcept] = None
    type: CodeableConcept
    source: Optional[Reference] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    calibration: Optional[List[DeviceMetricCalibration]] = field(default_factory=list)
    operational_status: Optional[str] = None